syntax = 'proto3';

package proto;

enum Enum {
	EnFlag1 = 0;
	EnFlag2 = 1;
}

message Message1 {
	int32 member1 = 8;
	int64 member2 = 2;
	uint32 member3 = 3;
	uint64 member4 = 4;
	Enum member5 = 7;
	bool member6 = 1;
	string member7 = 5;
	map<string, int32> member8 = 6;
	repeated int32 member9 = 9;
}

message Message2 {
	message Message3 {
		map<string, int64> member1 =2;
		repeated int64 member2 = 1;
	}
	map<string, Message3> member1 = 1;
	repeated Message1 member2 = 2;
	double member3 = 3;
	float member4 = 4;
	Message1 member5 = 5;
	Message3 member6 = 6;
}

